common/src/main/java/com/redhat/cloud/notifications/models/validation/NotPrivateHostnameValidator.java

```diff
@@ -0,0 +1,36 @@
+package com.redhat.cloud.notifications.models.validation;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.net.InetAddress;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.UnknownHostException;
+
+public class NotPrivateHostnameValidator implements ConstraintValidator<NotPrivateHostname, String>  {
```

Comment:
I have included the IPs in the test and the annotation seems to pick them up too. The same thing for the `AllowedHttpSchemeValidator`, since it should reject URLs without the `http` or `https` schemes specified. So I guess it validates the IPs too... however it will also not be used just by itself, so I was also thinking on maybe grouping these three annotations into one or something...

Reply:
I saw that you added tests for the rejected case for the ips - which i think it's fine. I'm guessing that it won't accept non internal ips? i.e. `34.235.198.240`